yapapi/services/service_state.py

```diff
@@ -11,31 +11,77 @@ class ServiceState(statemachine.StateMachine):
 
     # states
     pending = statemachine.State("pending", initial=True)
+    """The service instance has not yet been assigned to a provider."""
+
     starting = statemachine.State("starting")
+    """The service instance is starting on a provider.
+
+    The activity within which the service is running has been created on a provider
+    node and now  the service instance's :meth:`~yapapi.services.Service.start`
+    handler is active and has not yet finished.
+    """
+
     running = statemachine.State("running")
+    """The service instance is running on a provider.
+
+    The instance's :meth:`~yapapi.services.Service.start` handler has finished and
+    the :meth:`~yapapi.services.Service.run` handler is active.
+    """
+
     stopping = statemachine.State("stopping")
+    """The service instance is stopping on a provider.
+
+    The instance's :meth:`~yapapi.services.Service.run` handler has finished and
+    the :meth:`~yapapi.services.Service.shutdown` handler is active.
+    """
+
     terminated = statemachine.State("terminated")
+    """The service instance has been terminated and is no longer bound to an activity.
+
+    This means that either the service has been explicitly stopped by the requestor,
+    or the activity that the service had been attached-to has been terminated - e.g.
+    by a failure on the provider's end or as a result of termination of the agreement
+    between the requestor and the provider.
+    """
     unresponsive = statemachine.State("unresponsive")
 
     # transitions
-    start = pending.to(starting)
-    ready = starting.to(running)
-    stop = running.to(stopping)
-    terminate = terminated.from_(starting, running, stopping, terminated)
-    mark_unresponsive = unresponsive.from_(starting, running, stopping, terminated)
-    restart = pending.from_(pending, starting, running, stopping, terminated, unresponsive)
-
-    # transition performed when handler for the current state terminates normally,
-    # that is, not due to an error or `ControlSignal.stop`
+    start: statemachine.Transition = pending.to(starting)
+    """
+    Transition from :attr:`~ServiceState.pending` to :attr:`~ServiceState.starting`.
```

Comment:
Assuming we remove this from the API Reference (as I suggested in the first comment in this review), I think we should also remove this comment, and all the similar following comments - they are just spam from the `yapapi` developer POV.